module/mempool/consensus/exec_fork_suppressor.go

```diff
@@ -242,34 +242,22 @@ func (s *ExecForkSuppressor) RegisterEjectionCallbacks(callbacks ...mempool.OnEj
 	s.seals.RegisterEjectionCallbacks(callbacks...)
 }
 
-// enforceValidStates checks that seal has valid, non-empty, initial and final state.
+// enforceValidChunks checks that seal has valid non-zero number of chunks.
 // In case a seal fails the check, a detailed error message is logged and an
 // engine.InvalidInputError (sentinel error) is returned.
-func (s *ExecForkSuppressor) enforceValidStates(irSeal *flow.IncorporatedResultSeal) error {
+func (s *ExecForkSuppressor) enforceValidChunks(irSeal *flow.IncorporatedResultSeal) error {
 	result := irSeal.IncorporatedResult.Result
 
-	if _, ok := result.InitialStateCommit(); !ok {
-		scjson, err := json.Marshal(irSeal)
-		if err != nil {
-			return err
-		}
-		s.log.Error().
-			Str("seal", string(scjson)).
-			Msg("seal's execution result has no InitialStateCommit")
-		return engine.NewInvalidInputErrorf("seal's execution result has no InitialStateCommit: %x", result.ID())
-	}
-
-	if _, ok := result.FinalStateCommitment(); !ok {
-		scjson, err := json.Marshal(irSeal)
-		if err != nil {
-			return err
+	if !result.ValidateChunksLength() {
+		scjson, errjson := json.Marshal(irSeal)
+		if errjson != nil {
+			return errjson
```

Comment:
it seems to be a logic change here, needs @AlexHentschel attention

Reply:
Yep, @AlexHentschel  looked at it